tensorflow/core/kernels/bias_op_gpu.cu.cc

```diff
@@ -175,21 +174,34 @@ __global__ void BiasGradNCHW_SharedAtomics(const T* output_backprop,
   // Write the accumulated sum in this thread to the shared memory. Each thread
   // shifts their write location to avoid bank conflict.
   int bias_offset = threadIdx.x % 32;
-  CudaAtomicAdd(s_data + bias_offset, sum);
+  GpuAtomicAdd(s_data + bias_offset, sum);
   __syncthreads();
 
   // Accumulate the results in the shared memory into the first element.
   // No syncthreads is needed since this is only in the same warp.
   int32 thread_index = threadIdx.x;
+#if GOOGLE_CUDA
   if (thread_index < 32) {
     AccT data = s_data[thread_index];
     for (int32 delta = warpSize / 2; delta > 0; delta /= 2) {
-      data += CudaShuffleXorSync(kCudaWarpAll, data, delta);
+      data += GpuShuffleXorSync(kCudaWarpAll, data, delta);
     }
     if (thread_index == 0) {
-      CudaAtomicAdd(bias_backprop + bias_index, T(data));
+      GpuAtomicAdd(bias_backprop + bias_index, T(data));
     }
   }
+#elif TENSORFLOW_USE_ROCM
+  if (thread_index < 16) s_data[thread_index] += s_data[thread_index + 16];
```

Comment:
Would it make sense to implement this as a loop?

Reply:
probably.  Can we make that change as a follow-up PR?

We are in the process of debugging why the CUDA version of the implementation does not work on ROCm, and the idea is to file a follow-up PR with an implementation that works both on ROCm and CUDA platforms (or atleast have a better understanding of why it does not work).